ios/xpc/encoding.go

```diff
@@ -111,6 +112,28 @@ func EncodeData(w io.Writer, body map[string]interface{}) error {
 	return err
 }
 
+// TODO : remove this and figure out how to reuse EncodeData() to simulate what happens under this function
+func EncodeEmpty(w io.Writer, messageId uint64, additionalXpcFlags uint32, initHandshake bool) error {
```

Comment:
You could add a function `Encode(w io.Writer, m Message) error` (and also add `Id uint64` as a field to `Message`). Trying to derive the values for `Message` inside `EncodeData` doesn't make much sense anyways, the encoding part should only care about encoding

Reply:
Done, there is only one `EncodeMessage` function now.